### ISSUE TITLE
Unable to talk to daemon host using host's public ip

### ISSUE THREAD

**Issue by sburnwal** [moby/moby#25744]
I want a clarification. I have kept the docker0 bridge in a custom network like this (I am using docker 1.9.1).
/bin/docker daemon --bip=169.254.0.254/24 --fixed-cidr=169.254.0.0/24

And I created a network like this where I have put all all my containers.
docker network create --driver=bridge --subnet=169.254.1.0/24 --ip-range=169.254.1.0/25 --gateway=169.254.1.1 mynet

I have web server running on the local host at its (LAN) IP 172.23.166.174. From the container, I want to talk to web server using https://172.23.166.174:443/ but this is failing. Container is not able to reach the LAN ip of the host. Any other node in the same LAN like 172.23.166.175 or outside like www.google.com, it is able to reach fine. It is also able to communicate to other containers on the same mynet network.

I want to know whether this is allowed or not. If yes, when what am I missing ? Appreciate the reply, thanks.

**Comment by justincormack** [comment]
Is there anything on `169.254.1.1`? You set a gateway but it doesn't look like anything has that IP.

**Comment by sburnwal** [comment]
Some outputs from my host:

```
[root@pxgrid-106 ~]# ifconfig
br-3af76228f6f6: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 169.254.1.1  netmask 255.255.255.0  broadcast 0.0.0.0
        inet6 fe80::42:84ff:fec6:62c6  prefixlen 64  scopeid 0x20<link>
        ether 02:42:84:c6:62:c6  txqueuelen 0  (Ethernet)
        RX packets 247094  bytes 249809166 (238.2 MiB)
        RX errors 0  dropped 74  overruns 0  frame 0
        TX packets 35348  bytes 10680663 (10.1 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

docker0: flags=4099<UP,BROADCAST,MULTICAST>  mtu 1500
        inet 169.254.0.254  netmask 255.255.255.0  broadcast 0.0.0.0
        ether 02:42:02:74:f1:70  txqueuelen 0  (Ethernet)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 172.23.166.176  netmask 255.255.255.128  broadcast 172.23.166.255
        inet6 fe80::20c:29ff:fecc:7d0f  prefixlen 64  scopeid 0x20<link>
        ether 00:0c:29:cc:7d:0f  txqueuelen 1000  (Ethernet)
        RX packets 247094  bytes 249809166 (238.2 MiB)
        RX errors 0  dropped 74  overruns 0  frame 0
        TX packets 35348  bytes 10680663 (10.1 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 0  (Local Loopback)
        RX packets 2696372  bytes 969442185 (924.5 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 2696372  bytes 969442185 (924.5 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

veth7d58ac3: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet6 fe80::1cfc:69ff:fe8c:f37f  prefixlen 64  scopeid 0x20<link>
        ether 1e:fc:69:8c:f3:7f  txqueuelen 0  (Ethernet)
        RX packets 16008  bytes 1526226 (1.4 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 11593  bytes 2027148 (1.9 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

```

```
[root@pxgrid-106 ~]# ip route
default via 172.23.166.129 dev eth0 
169.254.0.0/24 dev docker0  proto kernel  scope link  src 169.254.0.254 
169.254.1.0/24 dev br-3af76228f6f6  proto kernel  scope link  src 169.254.1.1 
172.23.166.128/25 dev eth0  proto kernel  scope link  src 172.23.166.176 

```

**Comment by sburnwal** [comment]
Please let me know if you need any more info. So the bridge br-3af76228f6f6 has the IP 169.254.1.1. Do I still miss something ?

**Comment by sburnwal** [comment]
As of now, I need the first clarification - I know that in bridge mode, I can connect from the container to the docker0 ip on the host. I want to know whether I can connect to the public ip of the host as well in the bridge mode.

**Comment by sburnwal** [comment]
I am out of ideas what is the issue here with my linux - in bridge mode, container is not able to talk to the host's public ip. I had posted some outputs on the thread, see if they help.

I also took tcpdump on my host for container with ip 169.254.1.5 and host ip 172.23.166.176 port 443. Looks like tcp connection is never getting established successfully (see the attached screenshot) though some initial SYN and its ACK look ok but after that only 'TCP Spurious Retransmission' from the container to the host.
